src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -125,7 +134,20 @@ public async Task<IWorkspaceProjectContext> CreateProjectContextAsync(Guid id, s
             {
                 binOutputPath = data.GetRequiredPropertyAbsolutePathValue(BuildPropertyNames.TargetPath);
                 objOutputPath = GetIntermediateAssemblyPath(data, projectFilePath);
-                commandLineArgs = data.GetRequiredPropertyValue(BuildPropertyNames.CommandLineArgsForDesignTimeEvaluation);
+
+                // Property added in VS 17.4 compiler targets capturing values of LangVersion and DefineConstants.
+                // ChecksumAlgorithm value added to the property in 17.5.
+                //
+                // Impact on Hot Reload: incorrect ChecksumAlgorithm will prevent Hot Reload in detecting changes correctly in certain scenarios.
+                // However, given that projects that explicitly set ChecksumAlgorithm to a non-default value are rare and the project system
+                // will eventually call us to update the algorithm to the correct value, Hot Reload will likely not be impacted in practice.
+                commandLineArgs = data.GetPropertyValue(BuildPropertyNames.CommandLineArgsForDesignTimeEvaluation);
+
+                // Let EnC service known the checksum might not match, in case we need to diagnose related issue.
+                if (commandLineArgs.IsEmpty())
+                {
+                    _encService.LogPotentiallyMissingChecksumAlgorithm(_workspace.CurrentSolution.FilePath ?? "");
```

Comment:
This seems really icky that we have this directly depending on (and triggering a load of!) the ENC service here. How could we decouple things a bit better?

It also seems like the logic isn't right here either -- even if we didn't have this immediate property, we'd still get the checksum algorithm from the real design time build which would make E&C work again.

Reply:
We can move the set here. 

All that matters is that when we get a memory dump with VS Feedback and something is wrong with EnC I can check this and see if it is possible that the checksum alg was not read correctly.